### PR TITLE
refactor serverside of storing runs

### PR DESCRIPTION
**Pull request by gyorb**
* StorageSession was removed an upcoming patch will it make obsolate
  (used to check if there is an already running store for a run,
  this feature is removed with the cleanup)

* massStoreRun was refactored to multiple methods

### CODE REVIEW

**Comment by gyorb** [comment]
The run locking feature will be available with #1147 again.

File: libcodechecker/server/api/store_handler.py [hunk 1]
```diff
@@ -234,52 +231,42 @@ def addCheckerRun(session, storage_session, command, name, tag, username,
...
 
         run_history = RunHistory(run_id, tag, username, run_history_time)
         session.add(run_history)
-        session.flush()
 
-        storage_session.start_run_session(run_id, session)
+        session.flush()
         return run_id
     except Exception as ex:
-        session.close()
```
Comment (Xazax-hun) [inline libcodechecker/server/api/store_handler.py]: What is the reason this line being removed?

Comment (gyorb) [reply libcodechecker/server/api/store_handler.py]: The session will be closed automatically when the execution leaves the `DBSession` context. There is no need to close it there.

**Comment by whisperity** [comment]
@Xazax-hun Considering that there most likely won't be a release between the two patches, no functionality will be lost.

File: libcodechecker/server/api/report_server.py [hunk 2]
```diff
@@ -1827,19 +1754,168 @@ def getMissingContentHashes(self, file_hashes):
             return list(set(file_hashes) -
                         set(map(lambda fc: fc.content_hash, q)))
 
+    def __store_source_files(self, source_root, filename2hash):
```
Comment (whisperity) [inline libcodechecker/server/api/report_server.py]: Shouldn't we use `_to_` instead of `2`?

File: libcodechecker/server/api/report_server.py [hunk 3]
```diff
@@ -1827,19 +1754,168 @@ def getMissingContentHashes(self, file_hashes):
...
+        """
+        Storing file contents from plist.
+        """
+
+        file_path_to_id = {}
+
+        for file_name, file_hash in filename2hash.items():
```
Comment (whisperity) [inline libcodechecker/server/api/report_server.py]: Geeky wording here too.

File: libcodechecker/server/api/report_server.py (hunk 2 above)
Comment (whisperity) [inline libcodechecker/server/api/report_server.py]: If there is a file named `store_handler`, shouldn't these codes be moved there instead?

File: libcodechecker/server/api/report_server.py [hunk 4]
```diff
@@ -1865,180 +1949,33 @@ def massStoreRun(self, name, tag, version, b64zip, force):
...
-                                             # Round the duration to seconds.
-                                             int(sum(check_durations)))
+                                             durations)
 
-            # The database session initialized in addCheckerRun should be
-            # closed in finishCheckerRun after the report storage is done.
-            store_handler.finishCheckerRun(self.__storage_session, run_id)
+                store_handler.finishCheckerRun(session, run_id)
+
+                session.commit()
```
Comment (whisperity) [inline libcodechecker/server/api/report_server.py]: This is definitely the only part we `commit`, yes?

File: libcodechecker/server/api/store_handler.py (hunk 1 above)
Comment (whisperity) [reply libcodechecker/server/api/store_handler.py]: What are the commit rules for these? When `DBSession.__exit__`, is it just a `close()` without a commit, right?

File: libcodechecker/server/api/report_server.py (hunk 2 above)
Comment (gyorb) [reply libcodechecker/server/api/report_server.py]: I do not wanted to move it there so all of the database session creations are visible in the report server.

Comment (whisperity) [reply libcodechecker/server/api/report_server.py]: Understandable.

File: libcodechecker/server/api/report_server.py (hunk 4 above)
Comment (gyorb) [reply libcodechecker/server/api/report_server.py]: The massStore is split up to two main store sections, at first the source files are stored (each file in a new session). Everything after the source files are stored is done in one session (storing the reports), it should be successful and committed once or it should fail and rollback.

**Pull request merged**